File: leetcode/2960-count-tested-devices-after-test-operations/countTestedDevices.test.ts
import cases from "./cases.json";
import countTestedDevices from "./countTestedDevices";

test("count tested devices after test operations", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = countTestedDevices(
      testCase.input
    );
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toEqual(testCase.output);
  });
});

File: leetcode/2960-count-tested-devices-after-test-operations/countTestedDevices_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type CountTestedDevicesCase struct {
	Input []int

	Output int
}

func TestCountTestedDevices(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]CountTestedDevicesCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := countTestedDevices(testCase.Input)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}
